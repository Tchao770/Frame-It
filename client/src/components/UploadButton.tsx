import React from "react";
import { useEffect, useState } from "react";
import {
  Image,
  Platform,
  StyleSheet,
  Text,
  TouchableOpacity,
} from "react-native";
import * as ImagePicker from "expo-image-picker";
import { ImageObj, ImageProp } from "../types/data";

const UploadButton = ({ setImage }: ImageProp) => {
  useEffect(() => {
    async () => {
      if (Platform.OS !== "web") {
        const { status } =
          await ImagePicker.requestMediaLibraryPermissionsAsync();
        if (status !== "granted") {
          alert("Sorry we need camera roll permissions!");
        }
      }
    };
  }, []);

  const pickImage = async () => {
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.Images,
      /*
      allowsEditing: true,
        - iOS: Only Resizing, no cropping
        - Android: Everything.
      */
      quality: 1,
    });
    //console.log(result);
    if (!result.cancelled) {
      let resultObj: ImageObj = {
        uri: result.uri,
        height: result.height,
        width: result.width,
      };
      setImage(resultObj);
    }
  };

  return (
    <TouchableOpacity onPress={pickImage} style={styles.uploadContainer}>
      <Text style={styles.uploadText}>Upload</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  uploadContainer: {
    height: 70,
    width: 70,
    borderRadius: 50,
    backgroundColor: "dodgerblue",
    alignItems: "center",
    justifyContent: "center",
  },
  uploadText: {
    paddingBottom: 4,
  },
});

export default UploadButton;
