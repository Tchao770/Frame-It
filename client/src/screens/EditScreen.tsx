import React, { useRef, useEffect } from "react";
import {
  Button,
  Dimensions,
  Image,
  SafeAreaView,
  StyleSheet,
  Text,
  View,
} from "react-native";
import { ImageProp } from "../types/data";
import { manipulateAsync } from "expo-image-manipulator";

const MAX_WIDTH = Dimensions.get("window").width;
const MAX_HEIGHT = Dimensions.get("window").height;

const EditScreen = ({ image, setImage }: ImageProp) => {
  const rotateImage = async () => {
    const adjustedImage = await manipulateAsync(image.uri, [{ rotate: 90 }]);
    setImage(adjustedImage);
  };

  const resizeImage = async () => {
    const adjustedImage = await manipulateAsync(image.uri, [
      { resize: { width: image.width, height: image.height } },
    ]);
    setImage(adjustedImage);
  };

  const cropImage = async () => {
    const adjustedImage = await manipulateAsync(image.uri, [
      {
        crop: {
          originX: 0,
          originY: 0,
          width: image.width - 100,
          height: image.height - 100,
        },
      },
    ]);
    setImage(adjustedImage);
  };

  return (
    <SafeAreaView style={styles.editContainer}>
      <View style={styles.publishNav}>
        <Text>Confirm</Text>
      </View>
      <View style={styles.imageContainer}>
        {image && (
          <Image
            source={image.uri ? { uri: image.uri } : null}
            style={{
              resizeMode: "contain",
              maxWidth: MAX_WIDTH,
              maxHeight: MAX_HEIGHT,
              flex: 1,
            }}
          />
        )}
      </View>
      <View style={styles.editOptions}>
        <Button title="Rotate" onPress={rotateImage} />
        <Button title="Resize" onPress={resizeImage} />
        <Button title="Crop" onPress={cropImage} />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  editContainer: {
    width: "100%",
    height: "90%",
    backgroundColor: "gray",
    flexDirection: "column",
  },
  publishNav: { flex: 1, backgroundColor: "pink" },
  imageContainer: {
    flex: 5,
  },
  editOptions: {
    flex: 1,
    backgroundColor: "pink",
    flexDirection: "row",
    justifyContent: "space-around",
  },
});

export default EditScreen;
