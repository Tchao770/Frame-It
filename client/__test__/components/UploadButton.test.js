import React from "react";
import renderer from "react-test-renderer";
import UploadButton from "../../src/components/UploadButton";
import EditScreen from "../../src/screens/EditScreen";

describe("UploadButton Render", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<UploadButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("given an image selected from library, display it to the screen", () => {
    const { getByText } = render(<UploadButton />);
    fireEvent.press(getByText("Upload"));
    expect();
  });
});
