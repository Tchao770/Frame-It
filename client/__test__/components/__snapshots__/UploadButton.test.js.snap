// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton Render has 1 child 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "dodgerblue",
      "borderRadius": 50,
      "height": 70,
      "opacity": 1,
      "width": 70,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
